serial.c

```diff
@@ -84,21 +96,22 @@ static int check(enum sp_return result) {
 
   switch (result) {
   case SP_ERR_ARG:
-    SDL_LogError(SDL_LOG_CATEGORY_SYSTEM, "Error: Invalid argument.\n");
-    abort();
```

Comment:
You rarely want to call abort() to exit a program--that's an "abnormal" process exit like it crashed and it even dumps a crash log. According to the man page, it doesn't even flush the open streams. It would have been better to call exit() here since that causes a normal program exit. That's the standard way to exit a program when you want to.

But even better (in my opinion) it to return an error so that the higher level callers are notified of the error and shutdown more gracefully, so that's what I've done here.